syntax = "proto3";

import "msg_response.proto";
import "stream_data.proto";

package tfliteruntime;

message InitRequest{
    string model_path = 1;
    int32 num_threads = 2;
}

message SetInputRequest{
    oneof data{
        int32 index = 1;
        Chunk np_arr_chunk = 2;
    }
}

message InvokeRequest{}

message GetOutputRequest{
    int32 index = 1;
}

message GetOutputResponse{
    bytes np_data = 1;
}

// interface
service TfliteRuntimeServer{
    rpc Init(InitRequest) returns (MsgResponse) {}
    rpc SetInput (stream SetInputRequest) returns (MsgResponse) {}
    rpc Invoke (InvokeRequest) returns (MsgResponse) {}
    rpc GetOutput (GetOutputRequest) returns (stream GetOutputResponse) {}
}
